### ISSUE TITLE
Problem with overflow for dialog tooltips

### ISSUE THREAD

**Issue by beauwest** [Madelena/Metrology-for-Hass#6]
Hey! Tooltips for the entity detail dialogs are cut off instead of overflowing their container.

# Behavior
<img width="485" alt="Screen Shot 2022-05-10 at 8 13 08 AM" src="https://user-images.githubusercontent.com/193887/167636775-0b7e8824-dd06-4428-a9bf-ec5cac9ef9f9.png">

# Expected
<img width="439" alt="Screen Shot 2022-05-10 at 8 16 14 AM" src="https://user-images.githubusercontent.com/193887/167637276-473d2a38-078d-480d-90a3-a7a4a12dbc8a.png">

**Comment by Madelena** [comment]
Hmmmmmm I'm stumped on this one. After commenting out a bunch of codes to trace the bug, I've come to the conclusion that CSS backdrop-filter and filter breaks how it handles overflow of a block.

I did manage to fix it by disabling the overflow-y property of `.mdc-dialog__surface` and `.content`. But  the problem is that sometimes cards do overflow (e.g. when it has a lot of attributes) so I can't fix it this way...

![Weird bug](https://i.imgur.com/qsvcbEa.png)

If anyone had experience fixing this one, please let me know!

**Comment by beauwest** [comment]
I'm looking into it too, it has something to do with the `.mdc-dialog__surface::before`, but I can't figure out the differences yet. The `position: absolute` and `width/height: 100%` is supposed to be based on the entire page, but right now it's based on the dialog element.

I have to get to work, but if I have time, I'll dig into it more.

**Comment by Madelena** [comment]
The solution here will temporarily fix the scrollbar problem, but you lose the transparency blur in the process:

https://github.com/Madelena/Metrology-for-Hass/issues/7

**Comment by zanix** [comment]
I stumbled on a partial solution, it fixes overflow for tooltips while retaining the blur effect

![Tooltip Overflow - Light Mode](https://user-images.githubusercontent.com/1580378/186471045-5081a2c0-1466-4f35-827c-a0b24884507d.png)
![Tooltip Overflow - Dark Mode](https://user-images.githubusercontent.com/1580378/186471366-8fc2dd3e-ee0a-4ccc-ae8c-337ab5185488.png)

```yaml
  card-mod-more-info-yaml: &card-mod-more-info-yaml |
    $: |
      @supports (-webkit-backdrop-filter: none) or (backdrop-filter: none) {
        .mdc-dialog__surface {
          backdrop-filter: blur(16px);
          -webkit-backdrop-filter: blur(16px);
          background-color: rgba(var(--rgb-mdc-theme-surface), .5) !important;
        }
      }
      .mdc-dialog .mdc-dialog__container .mdc-dialog__surface {
        overflow: visible;
      }
      .mdc-dialog__surface {
        box-shadow: 0 0px 16px 4px rgba(var(--dialog-box-shadow), .15), 0 16px 64px 32px rgba(var(--dialog-box-shadow), .5) !important;
      }
```

However, dropdowns render incorrectly but it makes them more usable than before as noted in #28

![Dropdown - Dark Mode](https://user-images.githubusercontent.com/1580378/186471921-29cc8b36-e538-4b0f-8826-9f81fb081ff8.png)

I am no CSS expert, but if I remove the top and left style properties from this element, it returns to the proper position. I'm not sure how to apply this fix to the CSS though.

![Inspect](https://user-images.githubusercontent.com/1580378/186472642-851e5cd9-9051-4626-8aa8-e68f51126116.png)

![Position Fixed](https://user-images.githubusercontent.com/1580378/186472879-740bb235-f2f4-4c14-b377-6daf16e71a1e.png)

**Comment by Madelena** [comment]
Cool! Thank you and let me try this fix out.

**Comment by zanix** [comment]
Dropdowns seem to have been almost fixed with Home Assistant v2022.9.0 when using the overflow fix
![image](https://user-images.githubusercontent.com/1580378/189273980-9a595f5e-45d9-4f58-b3cb-d477c80a8b71.png)

Without the overflow fix, dropdowns cause scrollbars to show, but still show (almost) in the correct location
![image](https://user-images.githubusercontent.com/1580378/189274063-5daad29f-2bfc-4e27-8cc2-75b8a4d73207.png)

And the tooltips causing scrollbars still occur without the overflow fix
![image](https://user-images.githubusercontent.com/1580378/189274279-62fcdd7d-5754-4961-a150-284815e38d20.png)

**Comment by Madelena** [comment]
It's probably not an issue anymore since HA 2022.10 changed all the dialogs.

![image](https://user-images.githubusercontent.com/4341881/210485079-786bac87-ed9a-4c61-bab6-078d1869a498.png)

**Comment by zanix** [comment]
The overflow issue is still present for me on HA 2022.12
![image](https://user-images.githubusercontent.com/1580378/210487284-720f0e8c-fc4e-4003-89af-546e344cf89f.png)

**Comment by zanix** [comment]
Ah, I see my fix was added to v1.8

**Comment by mmalolepszy** [comment]
I had an issue with overflows and after manually applying a fix from @zanix it now looks like this:
![obraz](https://user-images.githubusercontent.com/9466870/210504376-843dea16-80e7-47b6-829b-b2d04b14f9c6.png)

What can I do about it?

I'm on HA 2022.12.9 and Metrology 1.7.2 installed via HACS.
